HabboHotel/Groups/GroupManager.cs

```diff
@@ -169,4 +171,34 @@ public List<Group> GetGroupsForUser(int userId)
         }
         return groups;
     }
+
+    public Dictionary<int, string>? GetHabboGroupBadges(Habbo habbo)
+    {
+        if (!habbo.GetClient().GetHabbo().InRoom)
+            return null;
+        var room = habbo.GetClient().GetHabbo().CurrentRoom;
+        if (room == null)
+            return null;
+        var badges = new Dictionary<int, string>();
+        foreach (var user in room.GetRoomUserManager().GetRoomUsers().ToList())
+        {
+            if (user.IsBot || user.IsPet || user.GetClient() == null || user.GetClient().GetHabbo() == null)
+                continue;
+            if (user.GetClient().GetHabbo().GetStats().FavouriteGroupId == 0 || badges.ContainsKey(user.GetClient().GetHabbo().GetStats().FavouriteGroupId))
+                continue;
+            if (!TryGetGroup(user.GetClient().GetHabbo().GetStats().FavouriteGroupId, out var group))
+                continue;
+            if (!badges.ContainsKey(group.Id))
+                badges.Add(group.Id, group.Badge);
+        }
+        if (habbo.GetClient().GetHabbo().GetStats().FavouriteGroupId > 0)

```

Comment:
If statement is obsolete. habbo is already part of the list of users iterated on line 183